cpp/src/quantiles/quantiles.cu

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+#include <cudf/copying.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/table/row_operators.cuh>
+#include <cudf/table/table_device_view.cuh>
+#include <cudf/types.hpp>
+#include <quantiles/quantiles_util.hpp>
+#include "cudf/utilities/traits.hpp"
+#include "thrust/functional.h"
+#include "thrust/iterator/counting_iterator.h"
+
+using ScalarResult = double;
+
+namespace cudf {
+namespace experimental {
+namespace detail {
+namespace {
+
+// enum class extrema {
+//     min,
+//     max
+// };
+
+// template<extrema minmax, bool is_nullable>
+// size_type extrema(column_view const & in,
```

Comment:
Uncommenting this code (and consuming it below) adds ~5m to compilation time for this file. The primary advantage is _probably_ space savings (no need to allocate a sorted index, since only a single location is returned). The index allocation should be temporary, though.